pylint/checkers/variables.py

```diff
@@ -370,6 +399,30 @@ def _ignored_modules(self):
         return get_global_option(self, 'ignored-modules', default=[])
 
 
+    @check_messages('reused-outer-loop-variable')
+    def visit_for(self, node):
+        assigned_to = _get_for_loop_assign_names(node.target)
+
+        # Only check variables that are used
+        dummy_rgx = self.config.dummy_variables_rgx
+        assigned_to = [var for var in assigned_to if not dummy_rgx.match(var)]
+
+        for variable in assigned_to:
```

Comment:
The `redefined-outer-name` check already finds this so I won't check for it here as well.